### ISSUE TITLE
unused variable in TranslationFromPretrainedBart.build_dataset_for_inference

### ISSUE THREAD

**Issue by sshleifer** [pytorch/fairseq#1918]
The `source_tokens` variable in
https://github.com/pytorch/fairseq/blob/bb1750d0442a617b08d7d727c049ab8c8e5cf460/fairseq/tasks/translation_from_pretrained_bart.py#L111

is created and then not used.

Is line 112 supposed to be 

```python
dataset = LanguagePairDataset(source_tokens, src_lengths, self.source_dictionary)
``` 
@yinhanliu

**Comment by yinhanliu** [comment]
I think you are right! Thanks for pointing this out. This is function has never been called in fairseq yet, since a pertained model is not able to translate anything. 
@ngoyal2707

**Comment by myleott** [comment]
Fixed in 9a718e29855713a51877237b2dcc25e39c234c82
